lte/gateway/python/magma/subscriberdb/store/base.py

```diff
@@ -116,6 +116,43 @@ def on_ready(self):
         """
         raise NotImplementedError()
 
+    @abc.abstractmethod
+    def add_apn_config(self, subscriber_data):
+        """
+        Method that should add the subscriber.
+
+        Args:
+            subscriber_data - SubscriberData protobuf message
+        Raises:
+            DuplicateApnError if the subscriber is already present
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    @contextmanager
```

Comment:
From Tariq: "why is this a contextmanager?
Also, "When abstractmethod() is applied in combination with other method descriptors, it should be applied as the innermost decorator""